test/supplemental_genbank_tests/genbank_upload_full_test.py

```diff
@@ -152,18 +148,252 @@ def test_simple_shock_upload(self):
         ### Test for upload from SHOCK - upload the file to shock first
         print('attempting upload through shock')
         gbk_path = "data/e_coli/GCF_000005845.2_ASM584v2_genomic.gbff"
-        data_file_cli = DataFileUtil(os.environ['SDK_CALLBACK_URL'])
         shutil.copy(gbk_path, self.cfg['scratch'])
-        shock_id = data_file_cli.file_to_shock({
+        shock_id = self.dfuClient.file_to_shock({
             'file_path': os.path.join(self.cfg['scratch'], gbk_path.split("/")[-1])
         })['shock_id']
         print("Running test")
         ws_obj_name2 = 'MyGenome.2'
-        result = self.getImpl().genbank_to_genome(self.getContext(), {
+        result = self.serviceImpl.genbank_to_genome(self.ctx, {
                 'file': {'shock_id': shock_id},
-                'workspace_name': self.getWsName(),
+                'workspace_name': self.wsName,
                 'genome_name': ws_obj_name2,
             })[0]
         self.assertIsNotNone(result['genome_ref'])
         self.assertTrue(int(
             result['genome_info'][10]['Number of Protein Encoding Genes']) > 0)
+
+    def _run_test_fail(self, params, error_message):
+        with pytest.raises(Exception) as got:
+            self.serviceImpl.genbank_to_genome(self.ctx, params)
+        assert_exception_correct(got.value, ValueError(error_message))
+
+    def _run_test_fail_mass(self, params, error_message):
+        with pytest.raises(Exception) as got:
+            self.serviceImpl.genbanks_to_genomes(self.ctx, params)
+        assert_exception_correct(got.value, ValueError(error_message))
+
+    def _check_result_object_info_fields_and_provenance(
+        self,
+        results,
+        file_names,
+        object_metas,
+        expected_provenance,
+    ):
+        object_version_pattern = re.compile(r'^[0-9]+\/1$')
+        for idx, res in enumerate(results):
+            assert object_version_pattern.match("/".join(res['genome_ref'].split("/")[-2:]))
+            data = self.wsClient.get_objects2({"objects": [{'ref': res['genome_ref']}]})["data"][0]
+
+            # check info
+            info = data["info"]
+            assert info == res['genome_info']
+            assert info[1] == file_names[idx]
+            assert info[2].split('-')[0] == 'KBaseGenomes.Genome'
+            assert info[6] == self.wsID
+            assert all(info[10].get(k) == v for k, v in object_metas[idx].items())
```

Comment:
It should be possible to do an exact test for everything other than:

3 - time (just parse to make sure it's a valid iso8601 string)
8 - md5, will change based on embedded upas and handles, ignore
9 - size, same as above

Reply:
👍

Reply:
:+1: